store/store.go

```diff
@@ -499,6 +506,67 @@ func (s *Store[H]) get(ctx context.Context, hash header.Hash) ([]byte, error) {
 	return data, nil
 }
 
+// notifyAndAdvance will notify waiters in heightSub and advance contiguousHead
+// based on already written headers.
+func (s *Store[H]) notifyAndAdvance(ctx context.Context, headers ...H) {
+	// always inform heightSub about new headers seen
+	for _, h := range headers {
+		s.heightSub.Notify(h.Height())
+	}
+
+	currHead := s.contiguousHead.Load()
+	if currHead != nil {
+		s.advanceContiguousHead(ctx, (*currHead).Height())
+	}
```

Comment:
Use Height from heightSub instead(on top of inlining comment above)

Reply:
Done

Reply:
Why do we need to keep contigiousHead if check?

Reply:
You mean `currHead != nil` ?

Reply:
Yes

Reply:
I dont think we need it

Reply:
I dont think we need it

Reply:
Looks like not. Maybe it works because of other change regarding first `flush`. I think before mentioned change we had a chance to meet nil `currHead`, hehe. Done.